cpp/src/join/mixed_join_kernels_semi.cu

```diff
@@ -69,8 +69,8 @@ __attribute__((visibility("hidden"))) __launch_bounds__(block_size) __global__
     auto equality = single_expression_equality<has_nulls>{
       evaluator, thread_intermediate_storage, false, equality_probe};
 
-    left_table_keep_mask[outer_row_index] =
-      hash_table_view.contains(outer_row_index, hash_probe, equality);
+    auto set_ref_ = set_ref.with_hash_function(hash_probe).with_key_eq(equality);
```

Comment:
we probably want to pass the custom key/hash on the host to save some time

Reply:
Resolved in [dc7b7d1](https://github.com/rapidsai/cudf/pull/16230/commits/dc7b7d1620c5fbebc24fe61a4aef89b03176d048)